crates/primitives/src/chain/spec.rs

```diff
@@ -1037,6 +1042,21 @@ impl From<Genesis> for ChainSpec {
         let time_hardfork_opts = [
             (Hardfork::Shanghai, genesis.config.shanghai_time),
             (Hardfork::Cancun, genesis.config.cancun_time),
+            #[cfg(feature = "optimism")]
+            (
+                Hardfork::Regolith,
+                genesis.config.extra_fields.get("regolithTime").and_then(|value| value.as_u64()),
+            ),
+            #[cfg(feature = "optimism")]
+            (
+                Hardfork::Ecotone,
+                genesis.config.extra_fields.get("ecotoneTime").and_then(|value| value.as_u64()),
+            ),
+            #[cfg(feature = "optimism")]
+            (
+                Hardfork::Canyon,
+                genesis.config.extra_fields.get("canyonTime").and_then(|value| value.as_u64()),
+            ),
```

Comment:
this works, but perhaps we could make this reusable by adding a helper struct that contains all the activation info as Optional fields and has a `OptimsismGenesisInfo::extract_from(&Genesis)` or similar

Reply:
sure done here https://github.com/paradigmxyz/reth/pull/7935/commits/1a781c34f29269d050f1195d71c16d89e7d836a7 PTAL